src/os/xdg.zig

```diff
@@ -193,3 +193,87 @@ test parseTerminalExec {
         try testing.expectEqualSlices([*:0]const u8, actual, &.{ "a", "-e", "b", "c" });
     }
 }
+
+/// https://specifications.freedesktop.org/basedir-spec/latest/
+pub const Dirs = enum {
+    config,
+    data,
+
+    pub fn key(self: Dirs) []const u8 {
+        return switch (self) {
+            .config => "XDG_CONFIG_DIRS",
+            .data => "XDG_DATA_DIRS",
+        };
+    }
+
+    pub fn default(self: Dirs) []const u8 {
+        return switch (self) {
+            .config => "/etc/xdg",
+            .data => "/usr/local/share:/usr/share",
+        };
+    }
+};
+
+pub const config_dirs = Dirs.config;
+pub const data_dirs = Dirs.data;
+
+pub const DirsIterator = struct {
+    data: []const u8,
+    iterator: std.mem.SplitIterator(u8, .scalar),
+
+    /// https://specifications.freedesktop.org/basedir-spec/latest/
+    pub fn init(env: std.process.EnvMap, key: Dirs) DirsIterator {
+        const data = data: {
+            if (env.get(key.key())) |data| {
+                if (std.mem.trim(u8, data, &std.ascii.whitespace).len > 0) break :data data;
+            }
+
+            break :data key.default();
+        };
+
+        return .{
+            .data = data,
+            .iterator = std.mem.splitScalar(u8, data, ':'),
+        };
+    }
+
+    pub fn next(self: *DirsIterator) ?[]const u8 {
+        return self.iterator.next();
+    }
+};
+
+/// https://specifications.freedesktop.org/basedir-spec/latest/
+pub fn dirsIterator(alloc: std.mem.Allocator, key: Dirs) std.mem.Allocator.Error!DirsIterator {
+    const env = std.process.getEnvMap(alloc);
```

Comment:
Allocating when we don't have to on systems like posix feels bad. We shouldn't allocate on posix.

Reply:
Allocations removed.